File: frontend/app/api/run/reference/1.c
#include <stdio.h>

int main() {
    int stack[100], n = 0, x;

    // Read first line of numbers
    while (scanf("%d", &x) == 1) {
        stack[n++] = x;
        int c = getchar();
        if (c == '\n' || c == EOF) break;
    }

    // Read second input (push)
    if (scanf("%d", &x) == 1) {
        stack[n++] = x;
    }

    // Print stack after push
    for (int i = 0; i < n; i++) printf("%d ", stack[i]);
    printf("\n");

    // Pop 2 elements
    if (n >= 2) n -= 2;
    else n = 0;

    // Print stack after pop
    for (int i = 0; i < n; i++) printf("%d ", stack[i]);
    printf("\n");

    return 0;
}

File: frontend/app/api/run/reference/1.py
def main():
    import sys

    # Read first line of numbers
    line1 = sys.stdin.readline().strip()
    stack = list(map(int, line1.split()))

    # Read second input (push)
    line2 = sys.stdin.readline().strip()
    if line2:
        x = int(line2)
        stack.append(x)

    # Print stack after push
    print(" ".join(map(str, stack)))

    # Pop 2 elements
    for _ in range(2):
        if stack:
            stack.pop()

    # Print stack after pop
    print(" ".join(map(str, stack)))


if __name__ == "__main__":
    main()
